src/infrastructure/implementations/blockstack-crux-user-repository.ts

```diff
@@ -0,0 +1,117 @@
+
+export interface IBlockstackCruxUserRepositoryOptions extends ICruxUserRepositoryOptions {
+    blockstackInfrastructure: ICruxBlockstackInfrastructure;
+    cacheStorage?: StorageService;
+    cruxDomain?: CruxDomain;
+}
+
+export class BlockstackCruxUserRepository implements ICruxUserRepository {
+    private cacheStorage?: StorageService;
+    private infrastructure: ICruxBlockstackInfrastructure;
+    private blockstackService: BlockstackService;
+    constructor(options: IBlockstackCruxUserRepositoryOptions) {
+        this.cacheStorage = options && options.cacheStorage;
+        const infrastructure = options.blockstackInfrastructure;
+        if (options.cruxDomain) {
+            const domainBnsOverrides = options.cruxDomain.config.nameserviceConfiguration ? options.cruxDomain.config.nameserviceConfiguration.bnsNodes : undefined;
+            infrastructure.bnsNodes = domainBnsOverrides && [...new Set([...infrastructure.bnsNodes, ...domainBnsOverrides])] || infrastructure.bnsNodes;
+            const gaiaHubOverride = options.cruxDomain.config.nameserviceConfiguration ? options.cruxDomain.config.nameserviceConfiguration.gaiaHub : undefined;
+            infrastructure.gaiaHub = gaiaHubOverride || infrastructure.gaiaHub;
+            const subdomainRegistrarOverride = options.cruxDomain.config.nameserviceConfiguration ? options.cruxDomain.config.nameserviceConfiguration.subdomainRegistrar : undefined;
+            infrastructure.subdomainRegistrar = subdomainRegistrarOverride || infrastructure.subdomainRegistrar;
+        }
+        this.infrastructure = infrastructure;
+        this.blockstackService = new BlockstackService({
+            bnsNodes: this.infrastructure.bnsNodes,
+            cacheStorage: this.cacheStorage,
+            subdomainRegistrar: this.infrastructure.subdomainRegistrar,
+        });
+        log.debug("BlockstackCruxUserRepository initialised");
+    }
+    public create = async (cruxId: CruxId, keyManager: IKeyManager): Promise<CruxUser> => {
+        const cruxUserInformation = await this.blockstackService.registerCruxId(cruxId, this.infrastructure.gaiaHub, keyManager);
+        return new CruxUser(cruxId, {}, cruxUserInformation);
+    }
+    public isCruxIdAvailable = async (cruxID: CruxId): Promise<boolean> => {
+        return this.blockstackService.isCruxIdAvailable(cruxID);
+    }
+    public getByCruxId = async (cruxID: CruxId, tag?: string, onlyRegistered: boolean = true): Promise<CruxUser|undefined> => {
+        const cruxUserInformation = await this.blockstackService.getCruxIdInformation(cruxID);
+        if (cruxUserInformation.registrationStatus.status === SubdomainRegistrationStatus.NONE) {
+            return;
+        }
+        let addressMap = {};
+        if (cruxUserInformation.registrationStatus.status === SubdomainRegistrationStatus.DONE) {
+            addressMap = await this.getAddressMap(cruxID, tag);
+        } else {
+            if (onlyRegistered) {
```

Comment:
@SanchayPahalwani 

This doesn't actually save any calls. Take `onlyRegistered` parameter inside blockstackService and block registrar call.

Make `onlyRegistered=false` default and override when calling.

Reply:
https://github.com/cruxprotocol/js-sdk/pull/151